.github/workflows/ci.yml

```diff
@@ -0,0 +1,53 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  POETRY_VERSION: 1.7.1
+  GHCR_IMAGE_REPOSITORY: ghcr.io/${{ github.repository_owner }}/poetry
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python:
+          - "3.8"
+          - "3.9"
+          - "3.10"
+          - "3.11"
+          - "3.12"
+        variant:
+          - bookworm
+          - slim
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build Docker Image
+        run: |
+          docker build --target test -t test-image \
+          --build-arg POETRY_VERSION=${{ env.POETRY_VERSION }} \
```

Comment:
We can just use the latest build push action with `push: false` and `target: test`.